### PR TITLE
adding swagger 2.0 swagger.json/yaml endpoint

### PR DESCRIPTION
**Pull request by KurtStam**

### CODE REVIEW

**Comment by msavy** [comment]
@EricWittmann CI failure seems to be the `SimpleEchoTest.after:43 expected:<9> but was:<8>` thing again. 

@KurtStam As I understand the code, this should essentially auto-generate a swagger 2 spec, which will reside at `/api` of the manager microservice? Correct?

**Comment by KurtStam** [comment]
Yes, it adds /swagger.json to the root context, for management it adds apiman/swagger.json and for the gateway api/swagger.json.

File: gateway/platforms/war/micro/pom.xml [hunk 1]
```diff
@@ -159,6 +159,18 @@
       <artifactId>jboss-servlet-api_3.0_spec</artifactId>
     </dependency>
     <dependency>
+      <groupId>io.swagger</groupId>
+      <artifactId>swagger-jaxrs</artifactId>
+      <optional>true</optional>
```
Comment (EricWittmann) [inline gateway/platforms/war/micro/pom.xml]: Why is this marked as optional?

Comment (KurtStam) [inline gateway/platforms/war/micro/pom.xml]: Was considering making it an optional feature but changed my mind later - feel free to remove it.

Comment (EricWittmann) [inline gateway/platforms/war/micro/pom.xml]: Thanks.  Already done.

**Pull request merged**